app/controllers/saml_idp_controller.rb

```diff
@@ -88,10 +88,14 @@ def redirect_to_verification_url
   end
 
   def profile_or_identity_needs_verification_or_decryption?
-    return false unless ial2_requested?
+    return false unless ial2_requested? || ialmax_requested_with_ial2_user?
```

Comment:
Just me out here, rallying against `unless`... and especially with a compound statement, I think it's really hard to process, I'd suggest breaking into two lines or flipping:

```suggestion
    return false if !ial2_requested? && !ialmax_requested_with_ial2_user?
```